### PR TITLE
gh-116938: Fix `dict.update` docstring and remove erraneous full stop from `dict` documentation

### PR DESCRIPTION
**Pull request by Prometheus3375**
Changes made:
1. Removed premature period introduced in #125213.
2. Added a similar update made for `MutableMapping.update` to `dict.update` docstring. Also changed `dict/iterable` to `mapping/iterable` as `E` can be indeed any mapping object, not just `dict`.

* Issue: gh-116938

----
📚 Documentation preview 📚: https://cpython-previews--125421.org.readthedocs.build/

### CODE REVIEW

**Review by skirpichev — APPROVED** [review]

**Review by methane — APPROVED** [review]

**Comment by Viicos** [comment]
Thanks @Prometheus3375 for catching these issues

**Comment by methane** [comment]
LGTM, but I don't merge this yet to ensure a chance for other core developers to review.
Ping me two week later if no other reviews.

**Comment by Prometheus3375** [comment]
@methane 2 weeks have passed, nobody else reviewed.

**Review by AA-Turner — APPROVED** [review]

**Pull request merged**
